urllib3/transport_security.py

```diff
@@ -28,11 +28,26 @@ def __init__(self, transport_security_store=None):
 
     def validate_connection(self, conn):
         """
-        Enforce connection security checks such as HSTS and HPKP.
+        Enforce connection security checks such as HSTS. This can be called
+        multiple times on the same connection before each request,
+        since new HSTS headers may be processed in the meantime.
+
         This is a stub, to be implemented later.
 
         :param conn:
             A :class:`urllib3.connection.HTTPConnection` instance.
+
+        """
+
+    def validate_ssl_connection(self, conn):
```

Comment:
`validate_new_connection` would be misleading, because per our earlier discussion, we actually call it every time we make a request (as we should, in case new state is added from processing response headers).

I like `validate_hpkp` and `validate_hsts`. They're clear and concrete, but they might discourage implementers from extending the class. One other thing that logically fits into TSM is manual fingerprint pinning (`assert_fingerprint`), concurrent with HPKP. But anyway, I'll go ahead and rename to these for now.